libdap2/dapodom.c

```diff
@@ -81,7 +81,7 @@ dapodom_print(Dapodometer* odom)
     if(odom->rank == 0) {
 	strlcat(line,"[]",sizeof(line));
     } else for(i=0;i<odom->rank;i++) {
-	sprintf(tmp,"[%lu/%lu:%lu:%lu]",
+	snprintf(tmp,sizeof(tmp),"[%lu/%lu:%lu:%lu]",
```

Comment:
It seems like the only reason to use snprintf() is in case of an incorrectly terminated string. In that case, it will prevent an overflow.

But not if you just use sizeof() on the original string. In that case, you will still get the exact same overflow. Or am I missing something?

Reply:
Let me flip it around: what is the reason to use `sprintf` instead of `snprintf` ?

Reply:
OK, now I get it.